src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -241,23 +247,25 @@ function makeStyleCoord(out: number[], zr: ZRenderType, appendToBody: boolean, z
 
 interface TooltipContentOption {
     /**
-     * `false`: the DOM element will be inside the container. Default value.
-     * `true`: the DOM element will be appended to HTML body, which avoid
-     *  some overflow clip but intrude outside of the container.
+     * Choose a DOM element which the tooltip element will be located in order to
+     * avoid some overflow clip but intrude outside of the container.
+     *
+     * this config can be either a DomElement, a function to choose a element
+     * or a selector string used by query delector to local a element
      */
-    appendToBody: boolean
+    appendTo: ((el?: HTMLElement) => HTMLElement | undefined | null) | HTMLElement | string
```

Comment:
Since I'm not sure what arguments are useful for the callback, just take your idea.
But the `el` can't explicitly show that it is the chart container and it's just a name that I use as the example, so let's rename it to `chartContainer`, or do you have any better name?

Reply:
Sure, I will do the rename.

Let me explain the usage of that input with more detail. If there is a better way to do so, please tell me.

In our project, we have a multi level dashboard(designed by user) and every dashboard can scroll insides like the picture below.
![image](https://user-images.githubusercontent.com/70554554/230266066-d5269fb1-ddff-44bd-a1fc-6c8446c568cf.png)

Also, we want to manage all common tooltip config for different charts in a single config file. And I cannot use `myChart.getDom()` in that config file. So I need to have that element as an input such than I can use `e => e.closest('.scroll-container')` to locale my tooltip element correctly.

Reply:
> Also, we want to manage all common tooltip config for different charts in a single config file. And I cannot use `myChart.getDom()` in that config file.

Looks reasonable to do so if you split the option and chart instance and thus can't call the chart API. But is it possible and practical to override the `appendTo` function in the chart instantiation block? Like this,

_**options.json**_
```json
{
  "xAxis": {
    "type": "category",
    "data": ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
  },
  "yAxis": {
    "type": "value"
  },
  "series": [
    {
      "data": [150, 230, 224, 218, 135, 147, 260],
      "type": "line"
    }
  ],
  "tooltip": {
    "trigger": "axis"
  }
}
```

_**chart.js**_
```js
const myChart = echarts.init(chartContainer)
const option = /* imported option JSON from somewhere */
// override the `appendTo` function
option.tooltip.appendTo = () => {
  const container = myChart.getDom()
  return container.closest('.scroll-container')
}
myChart.setOption(option)
```

It should be able to do the same job but might not look easy and elegant.

Reply:
You inspired me. Currently out code looks like 

`
import commonTooltipConfig from 'xx'
...

const option = {
  ...
  tooltip: {
     ...commonTooltipConfig,
     ...customTooltipConfigByChartType,
  },
}
`

I can change it to 

`
import getCommonTooltipConfig from 'xx'
...

const option = {
  ...
  tooltip: {
     ...getCommonTooltipConfig(myChartInstance),
     ...customTooltipConfigByChart
  },
}
`

And the config file can access the instance and we still can keep those common configs isolated.

So if you have strong willing to remove that input, I can remove it and do some refactor on my project.
It is up to you, I am ok with it.